stdlib/public/core/FloatingPointFromString.swift

```diff
@@ -0,0 +1,2525 @@
+      // Normal decimal
+      var significand = mostSignificantBitsFrom(work: work,
+                                                range: quotientRange,
+                                                count: targetFormat.significandBits,
+                                                remainderNonZero: remainderNonZero)
+      if significand >= (1 &<< targetFormat.significandBits) {
+        significand >>= 1
+        binaryExponent &+= 1
+      }
+      targetSignificand = significand
+      if binaryExponent > targetFormat.maxBinaryExponent {
+        return .infinity(sign: sign)
+      }
+    } else if binaryExponent >= targetFormat.minBinaryExponent &- targetFormat.significandBits {
+      // Subnormal decimal
+      let bitsNeeded = binaryExponent &- (targetFormat.minBinaryExponent &- targetFormat.significandBits)
+      binaryExponent = targetFormat.minBinaryExponent &- 1 // Flag as subnormal
+      let significand = mostSignificantBitsFrom(work: work,
+                                                range: quotientRange,
+                                                count: bitsNeeded,
+                                                remainderNonZero: remainderNonZero)
+      // Usually, overflowing the expected number of bits doesn't
+      // break anything; it just results in a significand 1 bit longer
+      // than we expected.
+
+      // Except when the significand overflows into the
+      // exponent.  Then we've transitioned from a subnormal to
+      // a normal, so the extra overflow bit will naturally get
+      // dropped, we just have to bump the exponent.
+      if significand >= (1 &<< (targetFormat.significandBits &- 1)) {
+        binaryExponent &+= 1
+      }
+      targetSignificand = significand
+    } else {
+      // Underflow
+      return .zero(sign: sign)
+    }
+  }
+
+  return .binary(significand: targetSignificand,
+                 exponent: Int16(truncatingIfNeeded: binaryExponent),
+                 sign: sign)
+}
+
+// ================================================================
+// ================================================================
+//
+// Float16
+//
+// ================================================================
+// ================================================================
+
+#if !((os(macOS) || targetEnvironment(macCatalyst)) && arch(x86_64))
+
+@available(SwiftStdlib 5.3, *)
+@c(_swift_stdlib_strtof16_clocale)
+@usableFromInline
+internal func _swift_stdlib_strtof16_clocale(
+  _ cText: Optional<UnsafePointer<CChar>>,
+  _ output: Optional<UnsafeMutablePointer<Float16>>
+) -> Optional<UnsafePointer<CChar>>
+{
+  // FIXME: This function was added during the Float16 bringup, but Unlike the
+  // Float and Double versions of this function, it was never actually called
+  // from inline code.  Can we just drop the whole thing?
+  fatalError()
+}
+
+@available(SwiftStdlib 5.3, *)
+internal extension Float16 {
+  init?(_ span: Span<UInt8>) {
```

Comment:
@glessard @milseman Someday when I decide to pursue exposing this as a public entrypoint, should it use `UTF8Span` instead of `Span<UInt8>`?